### PR TITLE
Return the results as TypesModels instead of Arrays

### PR DESCRIPTION
**Pull request by nickurt**
Currently this package returns the results as a plain array, so it's not possible to use built-in Laravel-features a-la [sum](https://laravel.com/docs/5.5/collections#method-sum), [max](https://laravel.com/docs/5.5/collections#method-max), [min](https://laravel.com/docs/5.5/collections#method-min), etc.

With this Pull Request the results will be fill'd in their own (Response)Models (_I didn't used the Types/* ones, b/c that looks more RequestTypes then ResponseTypes/Entities/Models to me_), so it's possible to use [Collection-based functions](https://laravel.com/docs/5.5/collections). It's also possible to call functions (e.g. getStatusText() instead of StatusCode with the list-call) to get the [correct meaning](https://www.wefact.nl/wefact-hosting/apiv2/variabelen-lijst/) of it.

If you have suggestions/comments for another naming of Models or something else, let me know, I can change them.

This Pull Request is also not a breaking change, the current users can use it without any problems ...

### CODE REVIEW

**Comment by hyperized** [comment]
Thanks for your active participation!

I'm going to need a little time read this over and think about it, if you have more idea's keep them coming :).

**Comment by Caroga** [comment]
Hi @nickurt !
I was overlooking this PR as @hyperized asked me for some feedback and noticed you have extended the objects based on Eloquent's Model.
I fear this would create a misdirection on the usage of the DTO's you really want to create.

The misdirection I see is that extending on the Eloquent Model one would assume there are datatables included. We do not provide any migrations for those, also this would be unneeded as the goal for this PR is not to store or represent any data in a local datastore as mysql, but data accessed from another api. We have not database specific actions available on this store. 
Also, extending from the Eloquent Model would make presume using `Attachement::create();` as a valid call, but this would actually throw a (fatal?) exception.

Wouldn't it be better to extend on the Collection instead, which provides the Eloquent Model with the named functionality? 
Or, if you really wish to keep with the Eloquent possibilities, then go with something like https://github.com/jenssegers/model.

**Comment by nickurt** [comment]
Thanks for the fast replies Caroga & hyperized!

Yeah, using the whole Eloquent-part is indeed a little bit too much after-all. Personally, I have a more preference to gonna use jessegers/model instead of extending collections. If you guys agree with this, I will modify the pull-request to this.

File: src/Models/CreditInvoice.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+use Jenssegers\Model\Model;
+
+class CreditInvoice extends Model
+{
+    /**
+     * @return string
+     */
+    public function getStatusText()
+    {
+        switch($this->Status){
```
Comment (hyperized) [inline src/Models/CreditInvoice.php]: Could you do something like this instead:

```php
class CreditInvoice extends Model
{
    
    private $statusTexts = [
            0 => 'Nog niet ontvangen',
            1 => 'Nog niet betaald',
            2 => 'Deels betaald',
            3 => 'Betaald',
            8 => 'Creditfactuur',
            9 => 'Vervallen',
        ];
        
    
    /**
     * @return string
     */
    public function getStatusText()
    {
        if(!is_int ($this->Status)){
            throw \Exception('Status text is not an int.');
        }
        
        if(!in_array($this->Status, array_keys($this->statusTexts))){
            throw \Exception('Status not recognized.');
        }
        
        return $this->statusTexts[$this->Status];
    }
}
```

File: src/WefactAPI.php [hunk 2]
```diff
@@ -117,6 +117,23 @@ protected function sendRequest($controller, $action, $params)
             ];
         } else {
             $result = json_decode($curlResp, true);
+
+            if($result['action'] != 'list' || $result['currentresults'] > 0) {
+                $informationKey = last(array_keys($result));
+                $typeClass = '\\Hyperized\\Wefact\Models\\'.class_basename($this);
```
Comment (hyperized) [inline src/WefactAPI.php]: `\\Hyperized\\Wefact\Models\\` -> `\\Hyperized\\Wefact\\Models\\`

File: src/WefactAPI.php [hunk 3]
```diff
@@ -117,6 +117,23 @@ protected function sendRequest($controller, $action, $params)
             ];
         } else {
             $result = json_decode($curlResp, true);
+
+            if($result['action'] != 'list' || $result['currentresults'] > 0) {
```
Comment (hyperized) [inline src/WefactAPI.php]: What does this `$result['action'] != 'list' ||` section add? I'm a little confused with this line vs `if($result['action'] == 'list') {` on 125

**Comment by hyperized** [comment]
@nickurt please wait with the fixes, I want to run some tests and concepts with this first.

**Comment by hyperized** [comment]
Hi @nickurt,

Again, thank you for contributing, but we're going to go ahead and close this merge request. Not because it's wrong or unacceptable (we both want to use objects and models instead of arrays) but we want to refactor the codebase to include this as a complete feature (a v3.0 if you like) rather than this approach in which the types are simply duplicated.

**Comment by hyperized** [comment]
I created an issue for this: https://github.com/hyperized/wefact/issues/17

**Comment by nickurt** [comment]
No problem, glad you guys liked the idea. I will gonna improve this idea in the future (in my fork) aswell ...

**Pull request closed without merging**